core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -388,14 +388,19 @@ private[spark] class TaskSchedulerImpl(
       val execId = shuffledOffers(i).executorId
       val host = shuffledOffers(i).host
       val taskSetRpID = taskSet.taskSet.resourceProfileId
+      val prof = sc.resourceProfileManager.resourceProfileFromId(taskSetRpID)
+      val targetExecutorRpID = if (prof.isForTaskOnly) {
+        ResourceProfile.DEFAULT_RESOURCE_PROFILE_ID
+      } else {
+        taskSetRpID
```

Comment:
This seems to be error-prone. If people are not aware of `isForTaskOnly`, they might use the wrong `taskSetRpID`.

How about we assign `ResourceProfile._id` to `DEFAULT_RESOURCE_PROFILE_ID ` if `isForTaskOnly=true`?

Reply:
Thanks pointing this out. Will try to add a new `TaskResourceProfile` to process task only profiles to make the change more clear.

Reply:
Thanks pointing this out. Will try to add a new `TaskResourceProfile` to process task only profiles to make the change more clear.